pkg/tool/codehub/branch.go

```diff
@@ -0,0 +1,74 @@
+package codehub
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type BranchList struct {
+	Result struct {
+		Total    int `json:"total"`
+		Branches []struct {
+			Commit             Commit `json:"commit"`
+			Name               string `json:"name"`
+			Protected          bool   `json:"protected"`
+			DevelopersCanPush  bool   `json:"developers_can_push"`
+			DevelopersCanMerge bool   `json:"developers_can_merge"`
+			MasterCanPush      bool   `json:"master_can_push"`
+			MasterCanMerge     bool   `json:"master_can_merge"`
+			NoOneCanPush       bool   `json:"no_one_can_push"`
+			NoOneCanMerge      bool   `json:"no_one_can_merge"`
+		} `json:"branches"`
+	} `json:"result"`
+	Status string `json:"status"`
+}
+
+type Commit struct {
+	ID            string `json:"id"`
+	Message       string `json:"message"`
+	AuthorName    string `json:"author_name"`
+	CommittedDate string `json:"committed_date"`
+	CommitterName string `json:"committer_name"`
+}
+
+type Branch struct {
+	*Commit
```

Comment:
branch and commit are different things, don't mix them

Reply:
The interface returns commit data only